### PR TITLE
fixed: wrong preprocessor check for gcc7

### PR DESCRIPTION
**Pull request by akva2**
sorry about that. @bska @atgeirr

### CODE REVIEW

**Comment by akva2** [comment]
jenkins build this please

**Comment by akva2** [comment]
@bska i can't really decode what the tests wants to happen, it seems gcc8 filesystem eats a slash? it passed locally on gcc7.

**Comment by atgeirr** [comment]
I also must admit I do not understand the failing test, from a naive reading I would expect it to always fail: I would think that constructing a path with `/x/y/z///' would collapse the slashes (which is what seems to happen in the shell at least, emacs seems to treat it like the above should mean `/`).

**Comment by bska** [comment]
> i can't really decode what the tests wants to happen

This particular check is (intended) to verify that the `outputFileName()`&mdash utility function;selected by argument-dependent lookup&mdash;generates the expected output filenames (`CASE.F0123` and `CASE.01.F0123`, respectively) whether the "basename" ends in a period or not.  The `.01` case is to ensure that we don't strip off extensions too aggressively.

However the check is wrong.  It really should be using
```C++
const auto odir         = std::string{"/x/y/z///"};
const auto ext          = std::string{"F0123"};
const auto reduced_odir = std::string{"/x/y/z/"};
// ...
BOOST_CHECK_EQUAL(fname, reduced_odir + "CASE.F0123"); // or + "CASE.01.F0123"
```
In other words, GCC 8 is doing the right thing here (collapsing sequences of consequtive `/` characters down to one `/` character) while GCC 7 behaves like `boost::filesystem` and does not collapse such sequences.

In order to get this to pass on both GCC 7 and GCC 8 (or later), I suggest we update the `odir` definition to use a single trailing slash.  It'll reduce one aspect of the test, but the essential functionality will still be exercised.

**Comment by akva2** [comment]
jenkins build this please
